### PR TITLE
Fix admittance controller to work with panda_mpc

### PR DESCRIPTION
**Pull request by jk-ethz**
Fixes the admittance controller to support an inverted wrench and a controller damping.

### CODE REVIEW

File: moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp [hunk 1]
```diff
@@ -27,6 +27,11 @@ bool PathAdmittanceController::init(hardware_interface::JointStateInterface* hw,
   }
   ROS_INFO_STREAM("[PathAdmittanceController] Subscribing to wrench topic [" << wrench_topic << "]");
 
+  if (!controller_nh.param<bool>("wrench_inverted", wrench_inverted_, false)) {
+    ROS_WARN_STREAM("[PathAdmittanceController] Failed to parse wrench_inverted setting");
+    return false;
+  }
```
Comment (grizzi) [inline moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp:33]: Nitpick comment: more then "inverted" would be "opposite"

File: moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp [hunk 2]
```diff
@@ -135,41 +136,39 @@ void PathAdmittanceController::update(const ros::Time& time, const ros::Duration
   Eigen::Matrix3d R(q);
 
   // Update measured wrench and tracking errors
-  force_ext_ = Eigen::Vector3d(wrench_.wrench.force.x,
-                                 wrench_.wrench.force.y, 
-                                 wrench_.wrench.force.z);
+  force_ext_ = Eigen::Vector3d(!wrench_inverted_ ? wrench_.wrench.force.x : -wrench_.wrench.force.x,
+                               !wrench_inverted_ ? wrench_.wrench.force.y : -wrench_.wrench.force.y,
+                               !wrench_inverted_ ? wrench_.wrench.force.z : -wrench_.wrench.force.z);
```
Comment (grizzi) [inline moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp:141]: Maybe cleaner
```
force_ext_ = Eigen::Vector3d(wrench_.wrench.force.x, 
                             wrench_.wrench.force.y, 
                             wrench_.wrench.force.z)
if (wrench_inverted_) force_ext_ = -1.0 * force_ext_;
```

File: moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp [hunk 3]
```diff
@@ -135,41 +136,39 @@ void PathAdmittanceController::update(const ros::Time& time, const ros::Duration
...
   threshold(force_ext_, force_threshold_);
 
-  force_integral_ += force_ext_ * dt;
-  force_integral_ = force_integral_.cwiseMax(-force_integral_max_);
-  force_integral_ = force_integral_.cwiseMin(force_integral_max_);
-  Eigen::Vector3d delta_position =
-      Kp_linear_.cwiseProduct(force_ext_) + Ki_linear_.cwiseProduct(force_integral_);
-  Eigen::Vector3d delta_position_transformed = R * delta_position;
+  dpos_ddot_ = /*M_inv_ **/ (force_ext_ - Kd_linear_.cwiseProduct(dpos_dot_) - Kp_linear_.cwiseProduct(dpos_));
```
Comment (grizzi) [inline moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp:144]: Rather then commenting, keep an M_inv_ matrix somewhere which is the diagonal identity matrix.

File: moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp (hunk 2 above)
Comment (jk-ethz) [reply moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp:141]: True, I was already pre-optimizing (avoiding floating point multiplications in this case and boiling it down to simpler jump assembly instructions) but it's less readable like this. Probably a unary minus `force_ext_ = -force_ext_;` is even better.

**Pull request merged**